### PR TITLE
Developer Mode and send EMS Read Commands from WebUI - #2116

### PR DESCRIPTION
**Pull request by proddy**
- Developer Mode and send EMS Read Commands from WebUI
- Move Version check from Status to Settings and add a dedicated page
- Banded rows in WebUI dashboard
- Updated Unity tests
- Remove 'All Values' button from WebUI (for now)

### CODE REVIEW

**Comment by mattreim** [comment]
Update German translation:

```lua
DEVELOPER_MODE: 'Entwicklermodus'
```

**Comment by MichaelDvP** [comment]
What should the developer mode do? I can only see a checkbox in Web and bools in system, but nothing that checks this flag?

**Comment by proddy** [comment]
> What should the developer mode do? I can only see a checkbox in Web and bools in system, but nothing that checks this flag?

oops, forgot to save. When setting developer mode it'll show a button on the Web Log page where you can send a read command

**Comment by MichaelDvP** [comment]
Wow, this build is 200k bigger than before, i guess the stringstream class?

**Comment by proddy** [comment]
oh, that is not good. I saw stream was used in the Modbus code, so I assumed it was already loaded somewhere. I'll re-write that function.

File: src/system.cpp [hunk 1]
```diff
@@ -2012,4 +2015,63 @@ bool System::uploadFirmwareURL(const char * url) {
...
+    // extract <deviceID> <type ID> [offset] [length] from string
+    char * p;
+    char   value[10] = {0}; // null just in case
+
+    // make a copy so we can iterate
+    char data_args[EMS_MAX_TELEGRAM_LENGTH];
+    for (uint8_t i = 0; i < strlen(data); i++) {
+        data_args[i] = data[i];
+    }
+    data_args[strlen(data)] = '\0'; // make sure its terminated
```
Comment (MichaelDvP) [inline src/system.cpp:2029]: EMS_MAX_TELEGRAM_LENGTH seems not the logical., it's not a telegram.
the for.. copy can overflow if data is longer, why not using strlcpy?

Comment (proddy) [reply src/system.cpp:2029]: that is better indeed

**Pull request merged**